trie/stacktrie.go

```diff
@@ -32,8 +32,10 @@ var (
 
 // StackTrieOptions contains the configured options for manipulating the stackTrie.
 type StackTrieOptions struct {
-	Writer  func(path []byte, hash common.Hash, blob []byte) // The function to commit the dirty nodes
-	Cleaner func(path []byte)                                // The function to clean up dangling nodes
+	Writer          func(path []byte, hash common.Hash, blob []byte) // The function to commit the dirty nodes
+	Cleaner         func(path []byte)                                // The function to clean up dangling nodes
+	NoLeftBoundary  bool                                             // Flag whether the nodes on the left boundary are skipped for committing
+	NoRightBoundary bool                                             // Flag whether the nodes on the right boundary are skipped for committing
```

Comment:
Can't we have
```
leftBoundary []byte // left boundary (may be nil). Must be hex-encoded path to left boundary, if set. 
rightBoundary []byte // right boundary (may be nil). Must be hex-encoded path to left boundary, if set. 
```
And also have a `SetLeftBoundary(path []byte)` which we can invoke when we insert the first item?

Reply:
> I will try to response with some more logical words.

No that's fine, I understand the problem and what you are saying. But I don't see why we would need a `NoLeftBoundary` boolean: we could just let a `nil` border signify `NoLeftBoundary`. 

So in my suggestion: we would explicitly set the left boundary to the first element we feed it (or, if initiated with a proof, then we'd set it to the origin). And vice versa with right boundary.

Reply:
> sounds good to me.

Heh, just as I changed my mind, and made a PR (https://github.com/ethereum/go-ethereum/pull/28361) with `lastPath` in the stacktrie, thus necessitating having the booleans in the options :)